libsycl/.clang-format

```diff
@@ -0,0 +1 @@
+BasedOnStyle: LLVM
```

Comment:
FWIW I'd really encourage `IndentPPDirectives: AfterHash`. Especially config headers become really hard to read without that.

Reply:
We set it in libc++ becuase it makes things a lot more readable in `__config`. compiler-rt sets it as well.

Reply:
added https://github.com/llvm/llvm-project/commit/e901d3015eeb76e7216dba6f7383590b469af315